common/src/main/java/tachyon/network/protocol/RPCBlockReadRequest.java

```diff
@@ -33,17 +34,19 @@ public RPCBlockReadRequest(long blockId, long offset, long length) {
     mLength = length;
   }
 
+  @Override
   public Type getType() {
     return Type.RPC_BLOCK_READ_REQUEST;
   }
 
   /**
-   * Decode the input {@link ByteBuf} into a {@link RPCBlockReadRequest} object and return it.
+   * Decodes the input {@link ByteBuf} into a {@link RPCBlockReadRequest} object and returns it.
    *
    * @param in the input {@link ByteBuf}
    * @return The decoded RPCBlockReadRequest object
    */
   public static RPCBlockReadRequest decode(ByteBuf in) {
+    Preconditions.checkNotNull(in);
```

Comment:
After second look, I don't think it adds too much value, because we will see the NPE thrown from this method without this check. The eager null checking is more useful for the arguments passed in the class constructor, where we can catch it early and see the stack trace.